rfcs/033-lineage-graph.md

```diff
@@ -0,0 +1,103 @@
+
+Interaction Details:
+
+- By default, we will display 1 level (or depth) of the lineage. Users will be able to control the level by clicking Expand/Collapse via each node. Or simply use the filter to change the number of levels shown.
+- User will be able to Pane and Zoom, and also reset the whole graph.
+- Clicking any node will display the metadata details of that particular node in the sidebar.
+
+Important to note: We will not display the complete list of upstream/downstream tables. We will only display X number of tables by default based on their usage and will control the number by a config flag.
+
+See the Screenshots in [this folder](https://github.com/amundsen-io/rfcs/tree/lineage-graph/assets/033).
+
+## Reference-level Explanation (aka Technical Details)
+
+All the UI/UX explanations mentioned in the above section.
+
+This will change the lineage backend endpoints to include extra querystring parameter(s) for the filtering, 
+like `nodes_per_level (Shows the top X by usage)`. The rest of the endpoints will remain the same, with no change in the logic. 
+
+## Drawbacks
+
+> Why should we _not_ do this?
+
+> Please consider:
+> Implementation cost, both in term of code size and complexity
+> Integration of this feature with other existing and planned features
+> The impact on onboarding and learning about Amundsen
+> Cost of migrating existing Amundsen installations (is it a breaking change?)
+> If there are tradeoffs to choosing any path. Attempt to identify them here.
+
+## Alternatives
+
+> Why is this design the best in the space of possible designs?
+> What other designs have been considered and what is the rationale for not choosing them?
+> What is the impact of not doing this?
+
+## Prior art
+
+> Discuss prior art, both the good and the bad, in relation to this proposal. A few examples of what this can include are:
+
+> Does this feature exist in other data search applications and what experience have their community had?
+> For community proposals: Is this done by some other community and what were their experiences with it?
+> Papers: Are there any published papers or great posts that discuss this? If you have some relevant papers to refer to, this can serve as a more detailed theoretical background.
+
+> This section is intended to encourage you as an author to think about the lessons from other projects, provide readers of your RFC with a fuller picture. If there is no prior art, that is fine - your ideas are interesting to us whether they are brand new or if it is an adaptation from other projects.
+
+## Unresolved questions
+
+> What parts of the design do you expect to resolve through the RFC process before this gets merged?
+> What parts of the design do you expect to resolve through the implementation of this feature before stabilization?
+> What related issues do you consider out of scope for this RFC that could be addressed in the future independently of the solution that comes out of this RFC?
```

Comment:
I think we are missing a complete breakdown of the tasks to perform.
Maybe more details about the technology to use?

Reply:
I've added the technical details in the Technical Details section to make it clear.